### PR TITLE
Snack Machine Remake

### PR DESCRIPTION
**Pull request by scarwolff**
## Description

Re-Sprited and Changed all the Products in the Traditional Vending Machine. Less Space Station Gimmicky and makes use of pre-existing snacks in the code that werent really used. Resprited Cheesy Poofs and Added the Junk factor to all Pre-War Snacks.

Also resprited and renamed Chemmaster/Condimaster for future wastelander bar usage/realism.
## Motivation and Context

For all my life i was like, dayum ion remember eating epic swag admeme 4 no raisins in the fallout universe so it got capped.

## How Has This Been Tested?

Replaces all Existing Vending Machines and Everything works well.

## Screenshot

image

## Changelog (necessary)
:cl:
add: Added New Junk Food Machine Sprite
add: Fixed Junk Food Machine Functionality
add: New chemmaster/condimaster sprite
del: Removed Chef Vendor Option
balance: Made pre-war food junkfood
/:cl:

### CODE REVIEW

File: code/modules/vending/_vending.dm [hunk 1]
```diff
@@ -334,15 +332,6 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 			dat += "[(coin ? coin : "")][(bill ? bill : "")]&nbsp;&nbsp;<a href='byond://?src=[REF(src)];remove_coin=1'>Remove</a>"
 		else
 			dat += "<i>No money</i>&nbsp;&nbsp;<span class='linkOff'>Remove</span>"
-	if(istype(src, /obj/machinery/vending/snack))
```
Comment (Laus4Deus) [inline code/modules/vending/_vending.dm:337]: Why delete this?

File: code/modules/vending/_vending.dm [hunk 2]
```diff
@@ -71,8 +71,6 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
 	var/obj/item/coin/coin
 	var/obj/item/stack/spacecash/bill
 
-	var/dish_quants = list()  //used by the snack machine's custom compartment to count dishes.
```
Comment (Laus4Deus) [inline code/modules/vending/_vending.dm:74]: and dis¿

File: code/modules/vending/_vending.dm [hunk 3]
```diff
@@ -372,25 +361,6 @@ IF YOU MODIFY THE PRODUCTS LIST OF A MACHINE, MAKE SURE TO UPDATE ITS RESUPPLY C
...
-	usr.set_machine(src)
-
-	if((href_list["dispense"]) && (vend_ready))
-		var/N = href_list["dispense"]
-		if(dish_quants[N] <= 0) // Sanity check, there are probably ways to press the button when it shouldn't be possible.
-			return
-		vend_ready = 0
-		use_power(5)
-
-		dish_quants[N] = max(dish_quants[N] - 1, 0)
```
Comment (Laus4Deus) [inline code/modules/vending/_vending.dm:385]: ¿?

File: code/modules/vending/snack.dm [hunk 4]
```diff
@@ -1,92 +1,37 @@
...
 	refill_canister = /obj/item/vending_refill/snack
-	var/chef_compartment_access = "28" //ACCESS_KITCHEN
 
 /obj/item/vending_refill/snack
-	machine_name = "Getmore Chocolate Corp"
+	machine_name = "Pure Patriot Treats"
 	charges = list(12, 2, 0)//of 36 standard, 6 contraband
 	init_charges = list(12, 2, 0)
 
-/obj/machinery/vending/snack/attackby(obj/item/W, mob/user, params)
```
Comment (Laus4Deus) [inline code/modules/vending/snack.dm:23]: Don't remove assets 😠

File: code/modules/vending/_vending.dm (hunk 1 above)
Comment (scarwolff) [reply code/modules/vending/_vending.dm:337]: unused code that makes it so you can put custom food into the snack machine, didnt make sense when used in the wasteland and was never used in the base game anyway. (Having an empty chefs special on random pre-war snack machines dont make sense)

File: code/modules/vending/_vending.dm (hunk 2 above)
Comment (scarwolff) [reply code/modules/vending/_vending.dm:74]: screwed with the snack machine code, had to be removed in order for snacc to be normal vendor without having to replace all of them manually

File: code/modules/vending/_vending.dm (hunk 3 above)
Comment (scarwolff) [reply code/modules/vending/_vending.dm:385]: ties into above

File: code/modules/vending/snack.dm (hunk 4 above)
Comment (scarwolff) [reply code/modules/vending/snack.dm:23]: assets been changed into something more fallouty, this is just the code that makes it a custom space station chef food dispenser. (We have actual vending machines for that)

**Comment by Laus4Deus** [comment]
Okay! checks out!

**Pull request merged**